### DISCUSSION TITLE
How to customize site based on build-time env variables?

### DISCUSSION THREAD

**Discussion by zoul** [vercel/next.js#34058] [Help] [answered]
I have some env variables that customize our Next site – for example I don’t want to insert the Google Analytics tracking code into development / preview builds. Something like:

```typescript
{env.allowAnalytics && (
  <script /* load analytics tracking code */>
)}
```

This works fine during the first server render, but obviously fails on the client where there is no env to speak of. Is there a way to set up some “build-time flags” during build time and have them available on the client, too?

(Like what if I want my development version to have red background and production version to have green background based on build-time `process.env`?)

I know I could set up these flags in `getStaticProps` for each page and then store them in props (making them available on the client) – but that means doing it for each and every page, doesn’t it?

**Comment by tpetersons** [comment, marked as answer]
Hi,
You can use this to determine which environment is running: `process.env.NODE_ENV`.
When running `next dev`, it will say 'development' and 'production' for `next start` command.
Also this variable can be changed in package.json.

**Reply by zoul** [reply]
Thank you! But if the code then runs on the browser side, the environment is completely different. And I want to customize behaviour based on the Node env regardless of where the code run. In other words – I have some Node env vars to customize the build, and then want the client to behave differently depending on these env vars.

**Reply by tpetersons** [reply]
Have you checked this docs?
https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

**Reply by zoul** [reply]
D-oh! Somehow I have completely missed that. Sounds like exactly what I need. Will try. Thank you very much!
